app/controllers/items_controller.rb

```diff
@@ -395,11 +396,29 @@ def build_advanced_search(params)
 
     if params[:clause].present?
       @search = build_query(params)
+      @items = @search
       session[:result_ids] = @search.map(&:full_identifier)
     else
       @search = ItemSearchService.build_advanced_search(params, current_user)
+      @items = @search.hits.map(&:result)
       session[:result_ids] = @search.hits.map{|h|h.stored(:full_identifier)}
     end
+  end
+
+  def build_deletable_params(item, items)
+    item.bulk_deleteable[:countries] = bulk_deletable_relation(ItemCountry, Country, :country_id, items)
+    item.bulk_deleteable[:subject_languages] = bulk_deletable_relation(ItemSubjectLanguage, Language, :language_id, items)
+    item.bulk_deleteable[:content_languages] = bulk_deletable_relation(ItemContentLanguage, Language, :language_id, items)
+    item.bulk_deleteable[:data_categories] = bulk_deletable_relation(ItemDataCategory, DataCategory, :data_category_id, items)
+    item.bulk_deleteable[:data_types] = bulk_deletable_relation(ItemDataType, DataType, :data_type_id, items)
+  end
 
+  def bulk_deletable_relation(relation, associated_resource, associated_resource_id, items)
+    ids = relation.where(item_id: items.map(&:id))
+            .group_by(&associated_resource_id)
+            .select { |k,v| v.size == items.size }.keys
```

Comment:
What is this select for? I'm having trouble working it out

Reply:
this will filter out all the associations to the ones that are common to everything in the search, hence `size == items.size`
for example if the search is for author it could contain items with multiple countries etc, this will only allow removing just the countries that are all exactly the same

I'm kinda changing my mind as I'm typing to the answer to you, may be it's not needed